lib/galaxy/webapps/galaxy/api/history_contents.py

```diff
@@ -538,24 +538,52 @@ def prepare_collection_download(
     @router.post(
         "/api/histories/{history_id}/contents/{type}s",
         summary="Create a new `HDA` or `HDCA` in the given History.",
+        operation_id="history_contents__create_typed",
     )
+    def create_typed(
+        self,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+        history_id: DecodedDatabaseIdField = HistoryIDPathParam,
+        type: Optional[HistoryContentType] = Path(
+            title="Content Type",
+            description="The type of the history element to create.",
+            example=HistoryContentType.dataset,
+        ),
+        serialization_params: SerializationParams = Depends(query_serialization_params),
+        payload: CreateHistoryContentPayload = Body(...),
+    ) -> Union[AnyHistoryContentItem, List[AnyHistoryContentItem]]:
+        """Create a new `HDA` or `HDCA` in the given History."""
+        return self._create(trans, history_id, type, serialization_params, payload)
+
     @router.post(
         "/api/histories/{history_id}/contents",
         summary="Create a new `HDA` or `HDCA` in the given History.",
         deprecated=True,
+        operation_id="history_contents__create",
```

Comment:
```suggestion
        operation_id="history_contents_create",
```

or was that on purpose?

Reply:
It was on purpose - ``<snake case [controller/tag/noun]>__<snake case [operation/verb]>`` is what I did throughout for the new Tool Shed controllers in https://github.com/galaxyproject/galaxy/pull/15829.

Reply:
And if you look at the generated typescript schema - they do a similar thing to generate unique ids consistently. They separate parts of the ID with double underscores - they just do it more verbosely.

Reply:
Thanks! I did notice the double underscores in the Typescript schema after sending the comment, sorry. For my understanding, is this a functionally necessary change or just to have nicer ids in the schema?

Reply:
Likely not necessary, I added them when I realized I needed to generate these duplicated methods with different sets of parameters - good naming that makes it through to the schema felt like documentation of the differences.